### ISSUE TITLE
enum binding broken since 3.7.0

### ISSUE THREAD

**Issue by jknack** [jdbi/jdbi#1632]
Hi,

 Just upgraded to 3.11.1 and found enum binding is broken, when enum overrides a method

Works:
```
enum Letter {
  A
}

assertTrue(Letter.class.isEnum());
assertTrue(Letter.A.getClass().isEnum());
```

Don't work:
```
enum Letter {
  A {
    public String method() {
      return "...";
    }
  }

   public abstract String method();
}

assertTrue(Letter.class.isEnum());
assertTrue(Letter.A.getClass().isEnum()); <<< THIS NOW FAILS (don't get why this break)
```

Jdbi code that stopped work:

```
handle.createUpdate("... set letter=:letter")
  .bind("letter", Letter.A);
```

Stacktrace:
```
org.jdbi.v3.core.statement.UnableToCreateStatementException: No argument factory registered for 'DAILY' of qualified type tv.digicash.entities.vo.TimePeriod$1 [statement:"insert into billet_generation_reports (time_period_type, time_period_year_reference, time_period_reference, member_id, number_of_billets) values (:timePeriodType, :timePeriodYearReference, :timePeriodReference, :memberId, :numberOfBillets)", arguments:{positional:{}, named:{timePeriodYearReference:2019,timePeriodReference:346,numberOfBillets:1,timePeriodType:DAILY,memberId:0}, finder:[]}]

	at org.jdbi.v3.core.statement.ArgumentBinder.factoryNotFound(ArgumentBinder.java:174)
	at org.jdbi.v3.core.statement.ArgumentBinder.lambda$null$2(ArgumentBinder.java:141)
	at java.base/java.util.Optional.orElseThrow(Optional.java:397)
	at org.jdbi.v3.core.statement.ArgumentBinder.lambda$null$3(ArgumentBinder.java:141)
	at org.jdbi.v3.core.statement.ArgumentBinder.lambda$null$4(ArgumentBinder.java:142)
	at org.jdbi.v3.core.statement.ArgumentBinder.lambda$bindNamed$1(ArgumentBinder.java:100)
	at org.jdbi.v3.core.statement.ArgumentBinder.lambda$wrapExceptions$6(ArgumentBinder.java:153)
	at org.jdbi.v3.core.statement.ArgumentBinder.bindNamed(ArgumentBinder.java:103)
	at org.jdbi.v3.core.statement.ArgumentBinder.bind(ArgumentBinder.java:60)
	at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1661)
	at org.jdbi.v3.core.result.ResultProducers.lambda$getGeneratedKeys$4(ResultProducers.java:107)
	at org.jdbi.v3.core.result.ResultIterable.lambda$of$0(ResultIterable.java:54)
	at org.jdbi.v3.core.result.ResultIterable.one(ResultIterable.java:133)
```

jdbi 3.11.1
jdk: 8

**Comment by leaumar** [comment]
Interesting discussion: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6708424

**Comment by stevenschlansker** [comment]
Fixed in `3.12.0` - thanks for reporting, and please let us know if there's further issues.
